Modules/_sqlite/connection.c

```diff
@@ -808,26 +808,43 @@ static void _pysqlite_drop_unused_cursor_references(pysqlite_Connection* self)
     Py_SETREF(self->cursors, new_list);
 }
 
+#ifndef SQLITE_DETERMINISTIC
+#define SQLITE_DETERMINISTIC 0x800
+#endif
```

Comment:
Why do you have to define the constant in Python? Does it mean that SQLite doesn't support the flag? I would prefer to raise an exception, rather than using an hardcoded constant.

Reply:
Do you think an exception should raised if Python is compiled with SQLite version that doesn't support this feature but is runned with SQLite version that supports it?

Reply:
What happens if you pass 0x800 to a SQLite which doesn't know/support this flag? I expect an error.

Reply:
I checked it with SQLite 3.8.2 (the most ancient dockerized version I found). Function is created, but it's not marked as deterministic, so optimizations are not used, it's reflected in the added test.

Reply:
IMHO it doesn't make sense to raise exception here because if `deterministic` is requested, but created function is not marked as deterministic this only affects performance, but indeed, flag should not be passed if it's unsupported.